File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    count={i:0 for i in id_list} # 신고자: 신고한 이용자 중 정지된 이용자 수
    record={i:set() for i in id_list} # 신고된 이용자: {신고한 이용자 1, 신고한 이용자 2, ...}
    for i in range(len(report)): # record 기록 작성
        reporter,reported=report[i].split()
        record[reported].add(reporter)
        
    for reported,reporters in record.items(): # 신고 횟수 확인
        if len(reporters)>=k:
            for j in list(reporters):
                count[j]+=1
    answer=list(count.values())
    
    return answer

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
/*
user - {유저: 유저번호}
reportTable - 신고한 유저 * 신고된 유저 형식의 2차원 배열
reportUser - {유저: 유저가 신고한 유저 집합}
reportedCount - {유저: 신고당한 횟수}
*/

function solution(id_list, report, k) {
    var answer = Array(id_list.length).fill(0);
    var user = {};
    var reportTable = [];
    var reportUser = {};
    var reportedCount = {};
    for (i = 0; i < id_list.length; i ++) {
        user[id_list[i]] = i;
        reportTable.push(Array(id_list.length).fill(false));
        reportUser[id_list[i]] = [];
        reportedCount[id_list[i]] = 0;
    }
    
    for (i of report) {
        let tmp = i.split(" ");
        let reporter = tmp[0];
        let reported = tmp[1];
        
        if (reportTable[user[reporter]][user[reported]] === false) {
            reportedCount[reported] += 1;
            reportTable[user[reporter]][user[reported]] = true;
            reportUser[reporter].push(reported);
        }
    }
    
    for (i = 0; i < id_list.length; i ++) {
        if (reportedCount[id_list[i]] >= k) {
            for (j = 0; j < id_list.length; j ++) {
                if (reportTable[j][i] === true) {
                    answer[j] += 1;
                }
            }
        }
    }
    
    return answer;
}